### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by twofingerrightclick**
Halfway through the fileConfig

### CODE REVIEW

File: Configuration/MockConfig.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            {
+                return false;
+            }
+            else
+                return true;
+        }
+
+        public bool SetConfigValue(string name, string? value)
+        {
+            IConfigUtilities.CheckValidConfigInput(name, value);
```
Comment (twofingerrightclick) [inline Configuration/MockConfig.cs]: set is always a true operation unless input is bad? Trying to figure out how to use the bool.

File: SampleApp/Program.cs [hunk 2]
```diff
@@ -1,12 +1,32 @@
...
 
 namespace SampleApp
 {
-    class Program
+    class Program 
     {
         static void Main()
         {
-            Console.WriteLine("Hello World!");
+            Console.WriteLine("Current Enviroment Config Values: ");
```
Comment (twofingerrightclick) [inline SampleApp/Program.cs]: Is this what we are supposed to do here? This is code I copied from someone else to see how it worked.

**Comment by kenny2892** [comment]
On the FileConfigTests, you could make the testFilePath into a static variable since it never changes. Other than that, looks good.

**Comment by MarkMichaelis** [comment]
## Assignment 4 (PR 180): (85.5 % code coverage but no SampleApp.Tests project)

| Item                  | Points |
| --------------------- | ------ |
| Assignment compile but major pieces are missing, or there are significant problems. |  5 |
| The assignment contains only a few required elements or has significant problems. | 5 |

| The assignment misses several required elements. The solution contains errors. | 20 |
| --------------------- | ------ |
| EnvironmentConfig  (with vallation of null, empty string, or names with spaces or '=') | ✔ |
| * FileConfig (with vallation of null, empty string, or names with spaces or '=') | ✔ |
| Addressed all warnings appropriately | ❌ |
| Data driven testing with `DataRow` attribute | ✔ |
| Use MockConfig from SampleApp | ✔ |
| Implement IConfig in FileConfig and MockConfig | ✔ |
| Add code analysis to **SampleApp** address all warnings.  It is okay to appropriately disable rules such as  localization/culture rules. |  ❌ |
| Nullable enabled for all projects | ❌ |
| Code Review Done | ✔ |

| Testing | 15 |
| --------------------- | ------ |
| Unit test project fails to compile, most test fails, or contains significant problems. | 10 |
| Most assigned work is tested or contain some errors. | 5 |
| SampleApp Tested from SampleApp.Tests | ❌ |
| All assigned work is tested | 5 |

| Extra Credit |  |
| --------------------- | ------ |
| Given at the instructor’s discretion for good work beyond the minimum assigned work. | 0 |
|- Provide a means to iterate over all the settings on a provider rather than only pulling out specific ones by name.  | ✔ |
|- Refactor the test classes that test through `IConfig` into a base class so that the same tests can be used to test multiple. | ✔ |
|- Create an IConfigGroup that returns a set of setting based on wildcard filters (rather than just a single setting) | ✔ |
|- Provide a mock instance for `System.Console`.  (You do not need to test method invocations of `System.Console`.) | ✔ |
|- Create and implement an `IConfigAsync` interface (we are unlikely to cover async this quarter). | ✔ |

**Pull request closed without merging**